src/indexation/pht.cpp

```diff
@@ -193,6 +193,14 @@ void Pht::lookupStep(Prefix p, std::shared_ptr<int> lo, std::shared_ptr<int> hi,
                 IndexEntry entry;
                 entry.unpackValue(*value);
 
+                auto it = std::find_if(vals->cbegin(), vals->cend(), [&](const std::shared_ptr<IndexEntry>& ie) {
+                    return ie->value == entry.value;
+                });
```

Comment:
This is actually related to a wider problem which deserves to be indexed by an issue. There's a solution that applies on DHT layer.